### PR TITLE
Add page about common errors

### PR DESCRIPTION
**Pull request by CGenie**
I'm quite new to `haddock` and I wanted to add basic docs to a large codebase. After asking some basic questions on IRC, it was suggested I fix the `haddock` docs a bit so that others won't need to ask again.

### CODE REVIEW

**Comment by Kleidukos** [comment]
Pretty great addition, thanks a lot!

**Comment by Kleidukos** [comment]
@CGenie Hi! Could you rebase your PR on the current ghc-9.2 branch? :)

File: haddock-api/.hspec-failures [hunk 1]
```diff
@@ -0,0 +1 @@
+FailureReport {failureReportSeed = 630888988, failureReportMaxSuccess = 100, failureReportMaxSize = 100, failureReportMaxDiscardRatio = 10, failureReportPaths = []}
```
Comment (Kleidukos) [inline haddock-api/.hspec-failures:1]: You can remove this file from the commit :)

**Review by Kleidukos — CHANGES_REQUESTED** [review]

File: haddock-api/.hspec-failures (hunk 1 above)
Comment (CGenie) [reply haddock-api/.hspec-failures]: OK done.

**Review by Kleidukos — APPROVED** [review]
LGTM

**Pull request merged**